src/map_extras.cpp

```diff
@@ -1399,17 +1453,130 @@ static void mx_fumarole( map &m, const tripoint &abs_sub )
+    const int x = portal_location.x;
+    const int y = portal_location.y;
+
+    switch( rng( 5, 5 ) ) {
+        //Mycus spreading through the portal
+        case 1: {
+            m.add_field( portal_location, fd_fatigue, 3 );
+            fungal_effects fe( *g, m );
+            for( const auto &loc : g->m.points_in_radius( portal_location, 5 ) ) {
+                if( one_in( 3 ) ) {
+                    fe.marlossify( loc );
                 }
             }
+            //50% chance to spawn pouf-maker
+            m.place_spawns( GROUP_FUNGI_FUNGALOID, 2, x - 1, y - 1, x + 1, y + 1, 1, true );
+            break;
+        }
+        //Netherworld monsters spawning around the portal
+        case 2: {
+            m.add_field( portal_location, fd_fatigue, 3 );
+            for( int i = x - 5; i <= x + 5; i++ ) {
+                for( int j = y - 5; j <= y + 5; j++ ) {
+                    if( rng( 1, 9 ) >= trig_dist( x, y, i, j ) ) {
+                        if( one_in( 15 ) ) {
+                            m.place_spawns( GROUP_NETHER_PORTAL, 1, i, j, i, j, 1, true );
+                        }
+                    }
+                }
+            }
+            break;
+        }
+        //Several cracks in the ground originating from the portal
+        case 3: {
+            m.add_field( portal_location, fd_fatigue, 3 );
+            for( int i = 0; i < rng( 1, 10 ); i++ ) {
+                tripoint end_location = { rng( 0, SEEX * 2 - 1 ), rng( 0, SEEY * 2 - 1 ), abs_sub.z };
+                std::vector<tripoint> failure = line_to( portal_location, end_location );
+                for( auto &i : failure ) {
+                    m.ter_set( { i.x, i.y, abs_sub.z }, t_pit );
+                }
+            }
+            break;
+        }
+        //Radiation from the portal killed the vegetation
+        case 4: {
+            m.add_field( portal_location, fd_fatigue, 3 );
+            const int rad = 10;
+            for( int i = x - rad; i <= x + rad; i++ ) {
+                for( int j = y - rad; j <= y + rad; j++ ) {
+                    if( trig_dist( x, y, i, j ) + rng( 0, 3 ) <= rad ) {
+                        const tripoint loc( i, j, abs_sub.z );
+                        dead_vegetation_parser( m, loc );
+                        m.adjust_radiation( loc.x, loc.y, rng( 20, 40 ) );
+                    }
+                }
+            }
+            break;
+        }
+        //Lava seams originating from the portal
+        case 5: {
+            if( abs_sub.z <= 0 ) {
+                int x1 = rng( 0,    SEEX     - 3 ), y1 = rng( 0,    SEEY     - 3 ),
+                    x2 = rng( SEEX, SEEX * 2 - 3 ), y2 = rng( SEEY, SEEY * 2 - 3 );
+
+                // Pick a random cardinal direction to also spawn lava in
+                // This will make the lava a single connected line, not just on diagonals
+                std::vector<direction> possibilities;
```

Comment:
```suggestion
                static const std::array<direction, 4> possibilities = { { EAST, WEST, NORTH, SOUTH } };
```

Reply:
Don't we have 8 possible directions? Vector makes more sense.

Reply:
> Vector makes more sense.

Why? This is never changed at runtime and the number of elements is known at compile-time, so it might as well be an `array`. Note: this is `std::vector`, not the geometric vector.

> Don't we have 8 possible directions?

See comment above that code: it picks a *cardinal direction*.

Reply:
Ooooh I see.  Reading is fundamental.  And requires coffee.

Reply:
Fixed.